packages/devtools_app/lib/src/shared/environment_parameters/environment_parameters_external.dart

```diff
@@ -64,6 +64,18 @@ class ExternalDevToolsEnvironmentParameters
     // This should always return a null value for 3p users.
     return null;
   }
+
+  @override
+  GaLink? g3FlutterIdeRecommendationLink() {
+    // This should always return a null value for 3p users.
+    return null;
+  }
+
+  @override
+  GaLink? g3DartIdeRecommendationLink() {
+    // This should always return a null value for 3p users.
+    return null;
+  }
```

Comment:
how about instead having a single method `TextSpan recommendedDebuggers()` that we implement both externally and internally?

Reply:
Good idea, switched to that